model/src/main/kotlin/licenses/LicenseClassifications.kt

```diff
@@ -56,6 +56,10 @@ data class LicenseClassifications(
                     getOrPut(category) { mutableSetOf() } += license.id
                 }
             }
+
+            categories.forEach { category ->
```

Comment:
> Make the `Map` contain all category names as keys in order to simplify its use.

I guess the more important thing here is to be able to distinguish map look-ups with a non-existing category as the key vs. an existing category that has no licenses assigned.

Here's a more compact implementation suggestion:

    val licensesByCategory: Map<String, Set<SpdxSingleLicenseExpression>> by lazy {
        buildMap<String, MutableSet<SpdxSingleLicenseExpression>> {
            categorizations.forEach { license ->
                categoryNames.forEach { category ->
                    val licenses = getOrPut(category) { mutableSetOf() }
                    if (category in license.categories) licenses += license.id
                }
            }
        }
    }

Reply:
Also, please add a test-case in `LicenseClassificationsTest`'s `"licensesByCategory" should` test case that highlights that for existing but empty categorizations now an empty set instead of `null` is returned (above `"return null when querying the licenses for an unknown category")`.

Reply:
> I guess the more important thing here is to be able to distinguish map look-ups with a non-existing category as the key vs. an existing category that has no licenses assigned.

Is this a general remark, or are you saying the commit message is not detailed enough and must be changed?

Reply:
It's a general remark. But personally, I would have mentioned (only) that in the commit message. IMO it's clearer than saying "to simplify its use" without an example of how the use would have looked before and after the change.

TL;DR no strict need to change the commit message, just a suggestion.

Reply:
ok.